### PR TITLE
LPS-123054 if the redirection comes from a nonexistent 404 page, remove 404 attribute

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

File: portal-web/docroot/html/portal/layout.jsp [hunk 1]
```diff
@@ -16,8 +16,16 @@
 
 <%@ include file="/html/portal/init.jsp" %>
 
+<%
+Boolean is404Page = (Boolean)request.getAttribute("IS_404_PAGE");
```
Comment (adolfopa) [inline portal-web/docroot/html/portal/layout.jsp:20]: @AliciaGarciaGarcia @4lejandrito The parameter/variable name feels a bit off. What about using "layoutNotFound" or something similar?

Comment (4lejandrito) [reply portal-web/docroot/html/portal/layout.jsp:20]: @adolfopa I am already reviewing this.

Comment (AliciaGarciaGarcia) [reply portal-web/docroot/html/portal/layout.jsp:20]: The value is true if _layout.friendly.url.page.not.found_ is not empty, but the _layoutNotFound_ name is not valid...

The corner case that we are trying to is that
- The layout it's not found, the property _layout.friendly.url.page.not.found_ exists but the layout that is redirected is invalid (not exists) so on https://github.com/liferay-lima/liferay-portal/pull/1081/files#diff-8de9d78cf9232003a40dac7e2af521aa1ee3a6bca556bd9bb3224d9bdb6d8d5eR6698 we remove it in the next iteration.

if is a _layout.friendly.url.page.not.found_ > a 404 redirection, we do not show the status/error page, but any other idea, is welcomed

**Comment by 4lejandrito** [comment]
@adolfopa I propose another solution which does not need an extra attribute. Instead it cleans the SessionErrors when the 404 page is sucessfully processed.

See https://github.com/AliciaGarciaGarcia/liferay-portal/pull/59.

What do you think? I think the new solution is cleaner but I am not 100% sure.

Thanks!

File: portal-web/docroot/html/portal/layout.jsp [hunk 2]
```diff
@@ -16,8 +16,16 @@
...
+
+if (is404Page == null) {
+	is404Page = Boolean.FALSE;
+}
+%>
+
 <c:choose>
-	<c:when test="<%= SessionErrors.contains(request, NoSuchLayoutException.class) %>">
+	<c:when test="<%= SessionErrors.contains(request, NoSuchLayoutException.class) && !is404Page %>">
```
Comment (adolfopa) [inline portal-web/docroot/html/portal/layout.jsp:28]: @4lejandrito @AliciaGarciaGarcia I've checked the other PR, and none of them makes my heart warm :-)

Wouldn't it be enough to update this condition to something like:

```
<c:when test="<% ... && !Objects.equals(request.getRequestURI(), PropsValues.LAYOUT_FRIENDLY_URL_PAGE_NOT_FOUND) %>">
```

There will be corner cases and all that, but from the outside, it seems unnecessary to set context stuff (paramenter, session error, etc.) because when the current URL is the not found one, we always want to display it, no matter what. Isn't that so?

Comment (4lejandrito) [reply portal-web/docroot/html/portal/layout.jsp:28]: Hi @adolfopa we already tried that, it is the first thing I suggested to @AliciaGarciaGarcia. The problem is that at the point of rendering that JSP the request URI is no longer the one we want but `/portal/layout...`.

Comment (AliciaGarciaGarcia) [reply portal-web/docroot/html/portal/layout.jsp:28]: In my on-live tests _request.getRequestURI()_ , on the point where it loads, will never be  _PropsValues.LAYOUT_FRIENDLY_URL_PAGE_NOT_FOUND_
will be _/c/portal/layout_

**Comment by 4lejandrito** [comment]
As discussed, we will forward https://github.com/AliciaGarciaGarcia/liferay-portal/pull/59.

**Comment by 4lejandrito** [comment]
Actually https://github.com/liferay-lima/liferay-portal/pull/1133.
